src/zenml/integrations/aws/service_connectors/aws_service_connector.py

```diff
@@ -1371,6 +1372,22 @@ def _configure_local_client(
                     "aws_session_token"
                 ] = credentials.token
 
+            aws_credentials_path = os.path.join(
+                users_home, ".aws", "credentials"
+            )
+
+            # Ensure the .aws directory exists
+            os.makedirs(os.path.dirname(aws_credentials_path), exist_ok=True)
+
+            # Ensure the credentials file exists
+            if not os.path.isfile(aws_credentials_path):
+                with open(aws_credentials_path, "a"):
+                    pass  # The file is automatically closed when exiting the with block
+
+            # Set the appropriate permissions for the .aws directory and credentials file
+            os.chmod(os.path.dirname(aws_credentials_path), 0o700)
+            os.chmod(aws_credentials_path, 0o600)
```

Comment:
Are we sure this works on every OS?

Reply:
@safoinme FYI, this is the piece of code in the `awcli` package that does this same thing:

```
    def _create_file(self, config_filename):
        # Create the file as well as the parent dir if needed.
        dirname = os.path.split(config_filename)[0]
        if not os.path.isdir(dirname):
            os.makedirs(dirname)
        with os.fdopen(os.open(config_filename,
                               os.O_WRONLY | os.O_CREAT, 0o600), 'w'):
            pass
```

Reply:
@safoinme can you use the code I suggested here ?